applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -544,57 +602,57 @@ static void discover_source_cb(struct bt_conn *conn, struct bt_codec *codec, str
 
 	(void)memset(params, 0, sizeof(*params));
 
-	ret = headset_conn_index_get(conn, &conn_index);
+	ret = channel_index_get(conn, NULL, &channel_index);
 
 	if (ret) {
 		LOG_ERR("Unknown connection, should not reach here");
-	} else {
-#if (CONFIG_BT_VCS_CLIENT)
-		ret = ble_vcs_discover(conn, conn_index);
-		if (ret) {
-			LOG_ERR("Could not do VCS discover");
-		}
-#endif /* (CONFIG_BT_VCS_CLIENT) */
+		return;
+	}
 
-		ret = bt_audio_stream_config(conn, &audio_streams[configured_stream_count++],
-					     sources[conn_index].ep, &lc3_preset_nrf5340.codec);
-		if (ret) {
-			LOG_WRN("Could not configure stream");
-			return;
-		}
+	ret = next_free_stream_index_get(&stream_index);
+	if (ret) {
+		LOG_WRN("Failed to get free index");
+		return;
+	}
 
-		if (!group_created) {
-			for (int i = 0; i < ARRAY_SIZE(audio_streams); i++) {
-				group_params[i].stream = &audio_streams[i];
-				group_params[i].qos = &lc3_preset_nrf5340.qos;
-
-				/* Make the first stream be sink */
-				if (i == 0) {
-					group_params[i].dir = BT_AUDIO_DIR_SINK;
-					LOG_DBG("Sink stream: %p", (void *)&audio_streams[i]);
-				} else {
-					group_params[i].dir = BT_AUDIO_DIR_SOURCE;
-					LOG_DBG("Source stream: %p", (void *)&audio_streams[i]);
-				}
-			}
+	ret = bt_audio_stream_config(conn, &audio_streams[stream_index], sources[channel_index].ep,
+				     &lc3_preset_nrf5340.codec);
+	if (ret) {
+		LOG_WRN("Could not configure stream");
+		return;
+	}
 
-			ret = bt_audio_unicast_group_create(group_params, ARRAY_SIZE(audio_streams),
-							    &unicast_group);
-			if (ret) {
-				LOG_ERR("Failed to create unicast group: %d", ret);
-				return;
+	if (!group_created) {
+		for (int i = 0; i < ARRAY_SIZE(audio_streams); i++) {
+			group_params[i].stream = &audio_streams[i];
+			group_params[i].qos = &lc3_preset_nrf5340.qos;
+
+			/* Make the first stream be sink */
```

Comment:
How can a stream be a sink? A stream exists between a source and a sink.

Reply:
No, in CIS a sink is the stream going from gateway to headset, but a source is the stream going from headset to gateway.